code/datums/diseases/wizarditis.dm

```diff
@@ -87,32 +87,33 @@ STI KALY - blind
 
 
 /datum/disease/wizarditis/proc/teleport()
-	var/list/theareas = get_areas_in_range(80, affected_mob)
-	for(var/area/space/S in theareas)
-		theareas -= S
+	var/list/possible_areas = get_areas_in_range(80, affected_mob)
```

Comment:
Prob best to do the `is_teleport_allowed` check before you do anything. Saves performance

Reply:
went ahead and just checked the mobs Z for this at the beginning with a guard proc